ext/symengine/ruby_basic.c

```diff
@@ -32,28 +32,37 @@ VALUE cbasic_binary_op(VALUE self, VALUE operand2,
     sympify(operand2, cbasic_operand2);
 
     cresult = basic_new_heap();
-    cwfunc_ptr(cresult, this, cbasic_operand2);
-    result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL, cbasic_free_heap,
-                              cresult);
-    basic_free_stack(cbasic_operand2);
 
-    return result;
+    int error_code = cwfunc_ptr(cresult, this, cbasic_operand2);
+    if (error_code == 0) {
+        result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL,
+                                  cbasic_free_heap, cresult);
+        basic_free_stack(cbasic_operand2);
+        return result;
+    } else {
+        basic_free_stack(cbasic_operand2);
+        rb_raise(rb_eRuntimeError, "Runtime Error");
```

Comment:
What does this error signify? The error message should be more descriptive than "Runtime Error", which does not give much information about what's actually going wrong.